src/cpp/elf/config/config_elfwriter.h

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT
+// Copyright (C) 2025, Advanced Micro Devices, Inc. All rights reserved.
+
+#ifndef AIEBU_ELF_CONFIG_ELF_WRITER_H_
+#define AIEBU_ELF_CONFIG_ELF_WRITER_H_
+
+#include <elfwriter.h>
+
+namespace aiebu {
+
+class config_elf_writer: public elf_writer
+{
+  constexpr static unsigned char ob_abi = 0x46;
+  constexpr static unsigned char version = 0x02;
+public:
+  config_elf_writer(): elf_writer(ob_abi, version)
+  { }
+
+  std::vector<char>
+  process(std::vector<writer>& mwriter) override
+  {
+    std::vector<symbol> syms;
+    add_text_data_section(mwriter, syms);
+    if (syms.size())
+    {
+      ELFIO::string_section_accessor str = add_dynstr_section();
+      add_dynsym_section(&str, syms);
+      add_reldyn_section(syms);
+      add_dynamic_section_segment();
+    }
+    //std::string uuid = mwriter[0].get_metadata("kernel.config.uuid");
+    //if (!uuid.empty())
+    //  add_note(NT_XRT_UUID, ".note.xrt.kernel.config.uuid", uuid);
```

Comment:
Do you want to add a `elf_writer::process_common_helper()` function which is used by both class `elf_write` and class `config_elf_writer`? The `process() `virtual function can call this `process_common_helper()` and then handle the delta themselves. This will remove the code duplication.